### PR TITLE
hi

### PR DESCRIPTION
**Pull request by den-superman**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private String color;
+    private double square;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Figure.java:5]: you don't need a field for storing square, remove it

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public Circle(double radius) {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Circle.java:6]: you should pass color in the constructor and initialize it using Figure constructor, the same with other figures

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        this.colors = colors;
+    }
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public void setRandom(Random random) {
+        this.random = random;
+    }
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/ColorSupplier.java:23]: you don't need all these methods

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "Rectangle", "Right triangle", "Square"};
+
+    public String[] getFigures() {
+        return figures;
+    }
+
+    public void setFigures(String[] figures) {
+        this.figures = figures;
+    }
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:17]: you don't need all these, remove it

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        ColorSupplier colorSupplier = new ColorSupplier();
+
+        switch (figures[random.nextInt(figures.length)]) {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:22]: let's switch case will be dependent not on figures names but number from 0 to 4

File: src/main/java/core/basesyntax/Main.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:4]: remove empty line

File: src/main/java/core/basesyntax/Main.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:6]: create constant from 6

File: src/main/java/core/basesyntax/Main.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < 6; i++) {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:8]:
```suggestion
        for (int i = 0; i < figures.length; i++) {
```

File: src/main/java/core/basesyntax/Main.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+            if (i < 3) {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:9]:
```suggestion
            if (i < figures.length / 2) {
```

File: src/main/java/core/basesyntax/Main.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+                figureSupplier.getRandomFigure().printSquareInfo();
+            } else {
+                figureSupplier.getDefaultFigure().printSquareInfo();
+            }
+            System.out.println("");
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:14]: remove this line

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double leg1;
+    private double leg2;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/RightTriangle.java:5]:
```suggestion
    private double firstLeg;
    private double secondLeg;
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double side1;
+    private double side2;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Rectangle.java:5]: think about better naming

File: src/main/java/core/basesyntax/Main.java (hunk 10 above)
Comment (den-superman) [reply src/main/java/core/basesyntax/Main.java:14]: для более удобного разделения фигур при выводе была

File: src/main/java/core/basesyntax/RightTriangle.java (hunk 11 above)
Comment (den-superman) [reply src/main/java/core/basesyntax/RightTriangle.java:5]: почему так, моё ведь лаконичнее и воспринимаются вроде одинаково?

**Comment by den-superman** [comment]
я кстати правильно сейчас пушу, туда, куда нужно?

File: src/main/java/core/basesyntax/Area.java [hunk 13]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Area {
+    double getSquare();
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Area.java:5]: remove empty line

File: src/main/java/core/basesyntax/Draw.java [hunk 14]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Draw {
+    void printSquareInfo();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Draw.java:4]:
```suggestion
    void draw();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final int figuresAmount = 5;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:8]:
```suggestion
    private static final int FIGURES_AMOUNT = 5;
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(50),
+                        random.nextInt(50));
+            default:
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(50));
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE.toString(), 10);
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:29]: let's create constant from 10

File: src/main/java/core/basesyntax/Main.java [hunk 17]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        final int figuresToPrint = 6;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:5]: make a global constant from this one

File: src/main/java/core/basesyntax/Area.java [hunk 18]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Area {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Area.java:3]:
```suggestion
public interface AreaCalculator {
```

File: src/main/java/core/basesyntax/Area.java [hunk 19]
```diff
@@ -0,0 +1,6 @@
...
+    double getSquare();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Area.java:4]:
```suggestion
    double getArea();
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]
Answering to your question, no you should create a separate branch and push solution there

File: src/main/java/core/basesyntax/Main.java (hunk 17 above)
Comment (den-superman) [reply src/main/java/core/basesyntax/Main.java:5]: what for?

**Pull request closed without merging**